### PR TITLE
[PREVIEW] Feature/sscs 3806 - As an appellant I want to view my completed answers to the tribunal so that I can see what I have said

### PR DESCRIPTION
**Pull request by jozzhart**
https://tools.hmcts.net/jira/browse/SSCS-3806

Added question/answer submitted date and state
Updated question view
Refactor date display using nunjucks filter
Added global constants

### CODE REVIEW

File: app/server/controllers/question.ts [hunk 1]
```diff
@@ -9,13 +9,16 @@ function getQuestion(getQuestionService) {
     const questionId = req.params.questionId;
     try {
       const response = await getQuestionService(hearingId, questionId);
+
       const question = {
         hearingId,
         questionId,
         header: response.question_header_text,
         body: response.question_body_text,
+        state: response.answer_state,
```
Comment (jozzhart) [inline app/server/controllers/question.ts]: this should probably be ```answer_state``` to be consistent with task-list

File: app/server/views/question.html [hunk 2]
```diff
@@ -3,75 +3,95 @@
...
+      {% else %}
+
+        <div id="completed-answer">
+          <p class="govuk-!-font-weight-bold">{{i18n.question.completedAnswer.header}}</p>
+          <p>{{question.answer.value}}</p>
+          <p>{{i18n.question.completedAnswer.submittedText}}:
+            {{question.answer.datetime | date}}</p>
         </div>
+
+      {% endif %}
```
Comment (davet1985) [inline app/server/views/question.html]: Might be clearer if this was separated into two templates and included in appropriately?

File: app/server/views/task-list.html [hunk 3]
```diff
@@ -30,8 +30,8 @@ <h2 class="govuk-heading-m">
             <h2 class="govuk-heading-s">Questions from the tribunal</h2>
             <p>
                 <span id="deadline-status">{{ i18n.taskList.deadline[deadlineExpiryDate.status] }}</span>
-                {% if deadlineExpiryDate.formatted %}
-                    <span id="deadline-date">{{ deadlineExpiryDate.formatted }}</span>.
+                {% if deadlineExpiryDate.expiryDate %}
+                    <span id="deadline-date">{{ deadlineExpiryDate.expiryDate | date }}</span>.
```
Comment (davet1985) [inline app/server/views/task-list.html]: Like it 👍

File: test/browser/2-question.test.ts [hunk 4]
```diff
@@ -113,7 +113,23 @@ describe('Question page', () => {
...
+      await taskListPage.clickQuestion(questionId);
+    });
+
+    it('displays the previously submitted answer', async() => {
+      const savedAnswer = await questionPage.getElementText('#completed-answer p:nth-child(2)');
+      expect(savedAnswer).to.equal('Another valid answer')
+    });
+
+    it('displays the previously submitted answer date', async() => {
+      const savedAnswerDate = await questionPage.getElementText('#completed-answer p:nth-child(3)');
```
Comment (davet1985) [inline test/browser/2-question.test.ts]: Might be worth putting a class or ID on this element, and in the test above?

File: test/mock/services/sumbit_answer.js [hunk 5]
```diff
@@ -3,6 +3,7 @@ const cache = require('memory-cache');
 
 function cacheAnswerState(questionId) {
   cache.put(`${questionId}.state`, 'submitted');
+  cache.put(`${questionId}.answer_datetime`, '2018-09-05T12:40:51Z');
```
Comment (davet1985) [inline test/mock/services/sumbit_answer.js]: I think this should just set the answer date to now?

Comment (jozzhart) [reply test/mock/services/sumbit_answer.js]: done

File: test/browser/2-question.test.ts (hunk 4 above)
Comment (jozzhart) [reply test/browser/2-question.test.ts]: added classes

**Pull request merged**